pkg/controller/template.go

```diff
@@ -39,6 +40,15 @@ var funcMap = gotemplate.FuncMap{
 		glog.Error("invalid type conversion on backendHash template function")
 		return ""
 	},
+	"hostnameRegex": func(hostname string) string {
+		rtn := regexp.MustCompile(`\.`).ReplaceAllLiteralString(hostname, "\\.")
+		rtn = regexp.MustCompile(`\*`).ReplaceAllLiteralString(rtn, "([^\\.]+)")
+		return "^" + rtn
```

Comment:
@from-nibly I'd change this to `"^" + rtn + "$"` so this doesn't match other hostnames that starts with this regex. Eg `*.xpto.com` would match `sub.xpto.com.br`. Problem with this change?